### PR TITLE
Remove SmartHab documentation

### PR DESCRIPTION
**Pull request by outadoc**
## Proposed change

This PR removes the SmartHab documentation, as the integration is being removed as well.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/67874
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/3289
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by frenck — CHANGES_REQUESTED** [review]
There is still an old redirect in place that needs to be handled:

`/components/smarthab /integrations/smarthab` (in `_redirects`).

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by MartinHjelmare** [comment]
Please link a brands repo PR that removes the smarthab assets too.

File: source/_redirects [hunk 1]
```diff
@@ -472,3 +472,5 @@
 /integrations/yessssms /more-info/removed-integration 301
 /integrations/yweather /more-info/removed-integration 301
 /components/zwave /integrations/zwave_js
+/components/smarthab /more-info/removed-integration 301
```
Comment (frenck) [inline source/_redirects]: We have updated the our redirect system last week, so we no longer need this one :)

```suggestion 
```

Comment (outadoc) [reply source/_redirects]: Great, thanks :)

**Review by frenck — APPROVED** [review]
Thanks, @outadoc 👍

**Pull request merged**